### DISCUSSION TITLE
Can the code for the content for other modules than HTML dialog be pre-rendered/filtered pre injection??

### DISCUSSION THREAD

**Discussion by ts-mz** [fancyapps/ui#118] [Q&A]
Hi again,

I have a problem using a consent compliance plugin ("complianz") for WordPress and the Vimeo video integration. ComplianZ is supposed to block the loading of external content unless the user expressly accepts the cookies connected to it and replace it with a placeholder, which, upon click, will accept the cookies and load the original content. 

However, since Fancybox is injected into the DOM on trigger, ComplianZ cannot rewrite the respective elements upon page load because the fancybox HTML doesn't exist at that point. Except in the case of the HTML dialog.

I've tried it by placing a vimeo video in the html dialog and calling it. The video in the dialog box will be blocked by Complianz, and are unblockable.

So I was wondering if it's possible to pre-render and hide the entire HTML code for the fanybox on page load and thus be able to block and replace the video embeds unless consent has been given.

GDPR compliance is very relevant in this case.

Alternatively, is there an accessible way to filter the pre-rendered JS-object before it is injected in the DOM?

Thanks again!

**Comment by fancyapps** [comment]
Hi,

It is currently not possible to pre-render or to reuse existing HTML code. Simply because there was no need for that and current approach allows to have any number of gallery items and only max 3 (current and next/prev) slides will be loaded at any time.

Sorry, I do not understand what do you mean by "to filter" or "pre-rendered JS-object".
